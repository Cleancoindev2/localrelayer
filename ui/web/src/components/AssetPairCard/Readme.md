Examples:

```js { "props": { "className": "example-wrapper" } }
  const { getTestToken } = require('../../utils/mocks');

  <div>
    <AssetPairCard token={getTestToken()} />
  </div>
```
