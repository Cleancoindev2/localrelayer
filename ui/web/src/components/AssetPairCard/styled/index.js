import styled from 'styled-components';
import {
  Card,
} from 'antd';


export const AssetPairCard = styled(Card)`
  height: 100%;
  margin: 0;

  .ant-card-body {
    padding: 15px;
  }
`;

export const AssetPairInfo = styled.div`
  display: flex;
  flex-direction: column;
`;

export const CardTitle = styled.div`
  font-size: 0.9rem;
  display: flex;
  justify-content: space-between;
`;

export const AssetPrice = styled.div`
  display: flex;
  font-size: 0.75rem;
`;

export const Volume = styled.div`
  display: flex;
  font-size: 0.75rem;
  flex-direction: column;
`;

export const Description = styled.div`
  font-size: 0.75rem;
`;

export const BaseAssetAddress = styled.a`
  font-size: 0.75rem;
  white-space: nowrap;
  max-width: 80px;
  overflow: hidden;
  text-overflow: ellipsis;
`;

export const LastPrice = styled.div`
  font-weight: 400;
  font-size: 0.9rem;
`;

export const PriceChange = styled.div`
  font-size: 0.9rem;  
`;

export const HighLowBlock = styled.div`
  display: flex;
  align-items: flex-start;
  width: 40%;
`;

export const ChangeVolumeBlock = styled.div`
  display: flex;
  flex-direction: column;
  align-items: flex-end;
  width: 60%;
`;
