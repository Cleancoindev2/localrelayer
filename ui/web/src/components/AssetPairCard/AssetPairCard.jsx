// @flow
import React from 'react';
import {
  Card,
} from 'antd';

import type {
  Node,
} from 'react';
import type {
  AssetPair,
} from 'instex-core/types';

import {
  ColoredSpan,
} from 'web-components/SharedStyledComponents';
import * as S from './styled';


type Props = {
  loading: boolean,
  assetPair: AssetPair,
};

const AssetPairCard = ({
  assetPair,
  loading = false,
}: Props): Node => {
  const assetDataA = assetPair?.assetDataA;
  const assetDataB = assetPair?.assetDataB;
  const tradingInfo = assetPair?.tradingInfo || {};
  const isPositive = (tradingInfo.change24Hour || 0) >= 0;
  return (
    <S.AssetPairCard loading={loading}>
      <Card.Meta
        title={(
          <S.CardTitle>
            <S.AssetPairInfo>
              {assetDataA?.assetData?.symbol}
              /
              {assetDataB?.assetData?.symbol}
              <S.BaseAssetAddress
                target="_blank"
                rel="noopener"
                href={`https://etherscan.io/token/${assetDataA?.assetData?.address}`}
              >
                {assetDataA?.assetData?.address}
              </S.BaseAssetAddress>
            </S.AssetPairInfo>
            <S.LastPrice>
              {tradingInfo?.lastPrice || 'No info about trades in 24hr'}
            </S.LastPrice>
          </S.CardTitle>
        )}
        description={(
          <S.AssetPrice>
            <div>
              <div>
                High:
                {tradingInfo.highPrice || '--'}
              </div>
              <div>
                Low:
                {tradingInfo.lowPrice || '--'}
              </div>
            </div>
            <div>
              <S.PriceChange>
                {isPositive
                  ? (
                    <ColoredSpan color="green">
                      {`+${tradingInfo.change24Hour || '0.00'}%`}
                    </ColoredSpan>
                  )
                  : (
                    <ColoredSpan color="red">
                      {`${tradingInfo.change24Hour || '0.00'}%`}
                    </ColoredSpan>
                  )}
              </S.PriceChange>
              <div>
                Volume:
                {tradingInfo.volume ? Number(tradingInfo.volume).toFixed(4) : 0}
              </div>
            </div>
          </S.AssetPrice>
        )}
      />
    </S.AssetPairCard>
  );
};

export default AssetPairCard;
