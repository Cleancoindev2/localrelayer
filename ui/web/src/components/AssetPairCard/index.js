// @flow
import AssetPairCard from './AssetPairCard';

export default AssetPairCard;

const layoutName = 'assetPairCard';
export const layout = {
  xs: {
    i: layoutName,
    x: 0,
    y: 0,
    w: 1,
    h: 3,
    static: true,
  },
  sm: {
    i: layoutName,
    x: 0,
    y: 0,
    w: 1,
    h: 3,
    static: true,
  },
  md: {
    i: layoutName,
    x: 0,
    y: 10,
    w: 3.5,
    h: 3,
    static: true,
  },
  lg: {
    i: layoutName,
    x: 0,
    y: 0,
    w: 3,
    h: 3,
    static: true,
  },
  lg2: {
    i: layoutName,
    x: 0,
    y: 0,
    w: 5,
    h: 3,
    static: true,
  },
  lg3: {
    i: layoutName,
    x: 0,
    y: 0,
    w: 2,
    h: 3,
    static: true,
  },
};
