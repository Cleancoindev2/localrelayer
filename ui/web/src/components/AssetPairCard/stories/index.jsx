import React from 'react';
import {
  storiesOf,
} from '@storybook/react';
import {
  withKnobs,
  boolean,
  object,
} from '@storybook/addon-knobs';

import TradingPageLayout from 'web-components/TradingPageLayout';
import AssetPairCard from '..';


const assetPair = {
  id: '0xe41d2489571d322189246dafa5ebde1f4699f498_0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2',
  assetDataA: {
    minAmount: '0',
    maxAmount: '10000000000000000000',
    precision: 5,
    assetData: {
      symbol: 'ZRX',
      address: '0xe41d2489571d322189246dafa5ebde1f4699f498',
    },
  },
  assetDataB: {
    minAmount: '0',
    maxAmount: '50000000000000000000',
    precision: 5,
    assetData: {
      symbol: 'WETH',
      address: '0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2',
    },
  },
  meta: {
    tradingInfo: {
      volume: '10001',
      change24Hour: -50,
      lastPrice: '5',
      highPrice: '10',
      lowPrice: '5',
    },
  },
};

const AssetPairCardStory = () => (
  <TradingPageLayout.Preview
    hideRest={boolean('Hide preview layout', false)}
    assetPairCard={(
      <AssetPairCard
        loading={boolean('Loading', false)}
        assetPair={object('assetPair', assetPair)}
      />
    )}
  />
);


/* Add note - loading should not be seen at all on th real app */
storiesOf('Components|AssetPairCardComponent', module)
  .addDecorator(withKnobs)
  .add(
    'default',
    AssetPairCardStory,
  )
  .add(
    'full screen',
    AssetPairCardStory,
    {
      options: {
        goFullScreen: true,
      },
    },
  );
