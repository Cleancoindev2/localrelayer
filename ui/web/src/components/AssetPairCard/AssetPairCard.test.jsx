import React from 'react';
import renderer from 'react-test-renderer';

import AssetPairCard from './AssetPairCard';
import {
  assetPair,
} from './stories';

test('AssetPairCard component', () => {
  expect(
    renderer.create(
      <AssetPairCard assetPair={assetPair} />,
    ).toJSON(),
  ).toMatchSnapshot();

  /* Empty */
  expect(
    renderer.create(
      <AssetPairCard />,
    ).toJSON(),
  ).toMatchSnapshot();
});
