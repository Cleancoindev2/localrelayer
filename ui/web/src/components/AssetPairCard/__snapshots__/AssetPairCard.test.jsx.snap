// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssetPairCard component 1`] = `
<div
  className="ant-card styled__AssetPairCard-fxyu4t-0 jjwmsN ant-card-bordered"
>
  <div
    className="ant-card-body"
    style={Object {}}
  >
    <div
      className="ant-card-meta"
    >
      <div
        className="ant-card-meta-detail"
      >
        <div
          className="ant-card-meta-title"
        >
          <div
            className="styled__CardTitle-fxyu4t-2 bzjkvR"
          >
            <div
              className="styled__AssetPairInfo-fxyu4t-1 iDTomX"
            >
              ZRX
              /
              WETH
              <a
                className="styled__BaseAssetAddress-fxyu4t-4 kPMFls"
                href="https://etherscan.io/token/0xe41d2489571d322189246dafa5ebde1f4699f498"
                rel="noopener"
                target="_blank"
              >
                0xe41d2489571d322189246dafa5ebde1f4699f498
              </a>
            </div>
            <div
              className="styled__LastPrice-fxyu4t-5 gpkfrA"
            >
              5
            </div>
          </div>
        </div>
        <div
          className="ant-card-meta-description"
        >
          <div
            className="styled__AssetPrice-fxyu4t-3 dYCxdi"
          >
            <div>
              <div>
                High:
                10
              </div>
              <div>
                Low:
                5
              </div>
            </div>
            <div>
              <div
                className="styled__PriceChange-fxyu4t-6 exsUWK"
              >
                <span
                  className="SharedStyledComponents__ColoredSpan-sc-1a8fz3a-0 btnrhe"
                  color="red"
                >
                  -50%
                </span>
              </div>
              <div>
                Volume:
                10001.0000
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`AssetPairCard component 2`] = `
<div
  className="ant-card styled__AssetPairCard-fxyu4t-0 jjwmsN ant-card-bordered"
>
  <div
    className="ant-card-body"
    style={Object {}}
  >
    <div
      className="ant-card-meta"
    >
      <div
        className="ant-card-meta-detail"
      >
        <div
          className="ant-card-meta-title"
        >
          <div
            className="styled__CardTitle-fxyu4t-2 bzjkvR"
          >
            <div
              className="styled__AssetPairInfo-fxyu4t-1 iDTomX"
            >
              /
              <a
                className="styled__BaseAssetAddress-fxyu4t-4 kPMFls"
                href="https://etherscan.io/token/undefined"
                rel="noopener"
                target="_blank"
              />
            </div>
            <div
              className="styled__LastPrice-fxyu4t-5 gpkfrA"
            >
              No info about trades in 24hr
            </div>
          </div>
        </div>
        <div
          className="ant-card-meta-description"
        >
          <div
            className="styled__AssetPrice-fxyu4t-3 dYCxdi"
          >
            <div>
              <div>
                High:
                --
              </div>
              <div>
                Low:
                --
              </div>
            </div>
            <div>
              <div
                className="styled__PriceChange-fxyu4t-6 exsUWK"
              >
                <span
                  className="SharedStyledComponents__ColoredSpan-sc-1a8fz3a-0 kSvotj"
                  color="green"
                >
                  +0.00%
                </span>
              </div>
              <div>
                Volume:
                0
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
